src/sentry/integrations/services/integration/impl.py

```diff
@@ -368,44 +373,43 @@ def send_incident_alert_notification(
         notification_uuid: str | None = None,
     ) -> bool:
         try:
-            sentry_app = SentryApp.objects.get(id=sentry_app_id)
-        except SentryApp.DoesNotExist:
-            logger.info(
-                "metric_alert_webhook.missing_sentryapp",
-                extra={
-                    "sentry_app_id": sentry_app_id,
-                    "organization_id": organization_id,
-                },
+            event = SentryAppEventType(
+                f"metric_alert.{INCIDENT_STATUS[IncidentStatus(new_status)].lower()}"
             )
+        except ValueError as e:
+            sentry_sdk.capture_exception(e)
             return False
 
-        metrics.incr("notifications.sent", instance=sentry_app.slug, skip_internal=False)
+        with SentryAppInteractionEvent(
+            operation_type=SentryAppInteractionType.PREPARE_WEBHOOK,
+            event_type=event,
+        ).capture() as lifecycle:
+            try:
+                sentry_app = SentryApp.objects.get(id=sentry_app_id)
+            except SentryApp.DoesNotExist as e:
+                sentry_sdk.capture_exception(e)
+                lifecycle.record_failure(e)
```

Comment:
Instead of raising we're capturing errors and recording fails, since this function is an RPC method. I talked a bit with @GabeVillalobos  about the purpose/behavior of RPC methods and they should be very binary (success or fail). We shouldn't care about the reason too much, plus if we raise an error, the method will retry up to 5 times (please correct me if I summarized this or interpreted the convo wrong). 

So if we run into errors like this where we don't anticipate a success on retry, we should not raise and instead return the end result.